### PR TITLE
Use the FxA android component

### LINKED ISSUE #1301: Authentication deadend
### [Description]:
I'm trying to login to the Android app. After I enter a 2fa code I get the page shown below, but there is no obvious way how to continue. There is no browser menu, since this an embedded webview inside the app and if I click the 'close' button I'm getting an authentication error message.

notes

### [Affected versions]:
This is tested on the Android app version 1.0android-c2791

### [Steps to reproduce]:
Just enable 2fa on your Firefox account and then try to login on the Notes Android app

### PR DESCRIPTION
**Pull request by carolkng**
Fixes #1301 
Blocked on https://github.com/mozilla/application-services/issues/128, https://github.com/mozilla-mobile/android-components/pull/489

ping @vladikoff

### CODE REVIEW

File: native/android/app/src/main/java/com/notes/FxaLoginActivity.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        RelativeLayout view = (RelativeLayout) findViewById(R.id.webviewLayout);
+        WebView webView = new WebView(getBaseContext());
+        view.addView(webView);
+
+        webView.getSettings().setJavaScriptEnabled(true);
+        webView.getSettings().setDomStorageEnabled(true);
+        CookieManager.getInstance().setAcceptCookie(true);
+
+        webView.clearCache(true);
```
Comment (vladikoff) [inline native/android/app/src/main/java/com/notes/FxaLoginActivity.java]: This doesn't seem to work, do you need to clear other stuff besides cache?

Comment (vladikoff) [reply native/android/app/src/main/java/com/notes/FxaLoginActivity.java]: @carolkng ^

File: native/android/app/build.gradle [hunk 2]
```diff
@@ -162,6 +162,11 @@ dependencies {
     // compile "com.android.support:appcompat-v7:23.0.1"
     compile "com.android.support:appcompat-v7:25.3.1"
     compile "com.facebook.react:react-native:+"  // From node_modules
+    compile "org.mozilla.components:fxa:0.16"
+}
+
+repositories {
+    mavenLocal()
```
Comment (pocmo) [inline native/android/app/build.gradle]: FYI: 0.16 has been released yesterday and can be pulled from jcenter now. :)

Comment (vladikoff) [reply native/android/app/build.gradle]: I guess there is no issue keeping this setting. I think if it cannot find it locally it will go to jcenter

**Comment by vladikoff** [comment]
Landed in https://github.com/mozilla/notes/pull/1310

**Comment by vladikoff** [comment]
Thank you so much @carolkng !

**Pull request closed without merging**